ghcide/src/Development/IDE/LSP/LanguageServer.hs

```diff
@@ -263,27 +263,30 @@ untilMVar mvar io = void $
     waitAnyCancel =<< traverse async [ io , readMVar mvar ]
 
 cancelHandler :: (SomeLspId -> IO ()) -> LSP.Handlers (ServerM c)
-cancelHandler cancelRequest = LSP.notificationHandler SCancelRequest $ \NotificationMessage{_params=CancelParams{_id}} ->
-  liftIO $ cancelRequest (SomeLspId _id)
+cancelHandler cancelRequest = LSP.notificationHandler SMethod_CancelRequest $ \TNotificationMessage{_params=CancelParams{_id}} ->
+  liftIO $ cancelRequest (SomeLspId (toLspId _id))
+  where toLspId :: (Int32 |? T.Text) -> LspId a
```

Comment:
Should be in `lsp`, perhaps? We define `LspId` there...

Reply:
Originally it was a candidate for a lsp package function, however it's only ever used this once, so I figured it's best just to leave it here.